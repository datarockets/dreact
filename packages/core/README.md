# `@dreact/core`

> TODO: description

## Usage

```
const core = require('@dreact/core');

// TODO: DEMONSTRATE API
```
