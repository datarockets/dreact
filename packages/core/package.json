{
  "name": "dreact",
  "version": "0.10.0",
  "description": "> TODO: description",
  "homepage": "https://github.com/datarockets/dreact#readme",
  "license": "ISC",
  "bin": {
    "dreact": "bin/dreact-bridge.sh"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/datarockets/dreact.git"
  },
  "scripts": {
    "test": "echo \"Error: run tests from root\" && exit 1"
  },
  "dependencies": {
    "next": "^10.2"
  }
}
