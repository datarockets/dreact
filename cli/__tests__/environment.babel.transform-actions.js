import { transformSync as babelTransform } from '@babel/core'
import prettier from 'prettier'

import plugin from '../environment/babel/transform-actions'

const transform = (src, options) => {
  const transformed = babelTransform(src, {
    plugins: [plugin],
    ...options,
  }).code

  return prettier.format(transformed, {
    filepath: __filename,
    singleQuote: true,
    semi: false,
    trailingComma: 'all',
    arrowParens: 'avoid',
  })
}

it('transforms empty new Action', () => {
  const input = 'const myActionCreators = new Action()'

  const result = transform(input, {
    filename: '/Project/src/collections/CollectionName/actions.js',
  })

  expect(result).toMatchSnapshot()
})

it('passes the first argument to transformed function', () => {
  const input = `
    const test = new Action({
      init: one => ({ two: one })
    })
  `

  const result = transform(input, {
    filename: '/Project/src/collections/CollectionName/actions.js',
  })

  expect(result).toMatchSnapshot()
})

it('does not change when outside of actions file', () => {
  const input = 'const myActionCreators = new Action()'

  const result = transform(input, {
    filename: '/Project/src/collections/CollectionName/saga.js',
  })

  expect(result).toMatchSnapshot()
})
