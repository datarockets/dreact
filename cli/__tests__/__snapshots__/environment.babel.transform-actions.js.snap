// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not change when outside of actions file 1`] = `
"const myActionCreators = new Action()
"
`;

exports[`passes the first argument to transformed function 1`] = `
"import createAction from 'dreact/helper-actions'
const test = createAction('CollectionName/test', {
  init: one => ({
    two: one,
  }),
})
"
`;

exports[`transforms empty new Action 1`] = `
"import createAction from 'dreact/helper-actions'
const myActionCreators = createAction('CollectionName/myActionCreators')
"
`;
